emulator/src/core/emulator/instructions/arithmetic.rs

```diff
@@ -88,6 +88,67 @@ impl Emulator {
         self.reg['a'] = result_byte;
         Ok(())
     }
+    
+    pub fn inx(&mut self, register: &str) -> EResult<()> {
+        let prev = self.reg[register];
+        self.reg[register] = prev.wrapping_add(1);
+        Ok(())
+    }
+    
+    pub fn dcx(&mut self, register: &str) -> EResult<()> {
+        let prev = self.reg[register];
+        self.reg[register] = prev.wrapping_sub(1);
+        Ok(())
+    }
+    
+    pub fn inr(&mut self, register: char) -> EResult<()> {
+        let prev: u8;
+        if register == 'm' {
```

Comment:
I feel like the two `if`s could be combined but I might be mistaken, not entirely sure what happens here.

Reply:
Not sure how